### ISSUE TITLE
Can the same id be returned as a reserved id for the same client

### ISSUE THREAD

**Issue by joswarmer** [LionWeb-io/specification#259]
Scenario:

- Client `X` asks for a series of reserved id's
- Server returns id `ID-1` and `ID-2`
- Client created a node with `ID-1`

Some time later
- Client `X` asks for a series of reserved id's

**Question:** can the server result include the still unused `ID-2`?

This is interesting, because if this is allowed, the repository can ensure (to some extent) that clients actually use reserved id's before getting new ones. 
It also is interesting, because a client may have _forgotten_ id's requested earlier.

**Comment by enikao** [comment]
Yes, we allow this explicitly.

From [bulk api spec](https://lionweb.io/specification/bulk/repo-access-api.html#ids):

> The repository [...]
> * MAY hand out the same unused ids to the same client more than once.

**Comment by joswarmer** [comment]
Right! I didn't notice that in the spec.  It does mean that if a client has already created a node with a reserved id locally (without storing it in the respository) it needs to be aware that newly requested id's can already be in use locally.
